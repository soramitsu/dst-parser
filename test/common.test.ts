import { parseStr } from './test_util'
import {
    trimEmptyLines, getCommonStartPadding,
    spaceSectionIsEqual
} from '../src/index'
import { describe, expect, it } from 'vitest'

describe('Common tests', () => {
    it('handles an empty string', () => {
      let parsed = parseStr('')
      expect(parsed).toMatchSnapshot()
    })

    it('can trim empty lines at the start, which may change indentation otherwise', () => {
        let lines = ['  ', '   ', 'Source code is expected to be here']
        let trimmed = trimEmptyLines(lines)
        expect(trimmed).toMatchSnapshot()
    })

    it('can trim empty lines at the end, which may change indentation otherwise', () => {
        let actual_text = 'Source code is expected to be here'
        let lines = [actual_text, '\t  ', '  \t']
        let trimmed = trimEmptyLines(lines)
        expect(trimmed).toMatchSnapshot()
    })

    it('can handle empty line lists that can not be trimmed', () => {
        let lines = []
        let trimmed = trimEmptyLines(lines)
        expect(trimmed).toMatchSnapshot()
    })

    it('can find a common start padding', () => {
        let lines = ['    a', '    b', '    c']
        let padding = getCommonStartPadding(lines)
        expect(padding).toMatchSnapshot()
        lines = ['    a', '  b', '   c']
        padding = getCommonStartPadding(lines)
        expect(padding).toMatchSnapshot()
    })

    it('can find a common start padding', () => {
        let lines = ['    a', '    b', '    c']
        let padding = getCommonStartPadding(lines)
        expect(padding).toMatchSnapshot()
        lines = ['    a', '  b', '   c']
        padding = getCommonStartPadding(lines)
        expect(padding).toMatchSnapshot()
    })

    it('can check that space padding is equal to a given value', () => {
        let lines = ['    a', '    b', '    c']
        expect(spaceSectionIsEqual(lines, 4)).toMatchSnapshot()
        lines = ['    a', '  b', '   c']
        expect(spaceSectionIsEqual(lines, 4)).toMatchSnapshot()
    })
})