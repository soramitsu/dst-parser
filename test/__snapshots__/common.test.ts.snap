// Vitest Snapshot v1

exports[`Common tests > can check that space padding is equal to a given value 1`] = `true`;

exports[`Common tests > can check that space padding is equal to a given value 2`] = `false`;

exports[`Common tests > can find a common start padding 1`] = `4`;

exports[`Common tests > can find a common start padding 2`] = `2`;

exports[`Common tests > can find a common start padding 3`] = `4`;

exports[`Common tests > can find a common start padding 4`] = `2`;

exports[`Common tests > can handle empty line lists that can not be trimmed 1`] = `[]`;

exports[`Common tests > can trim empty lines at the end, which may change indentation otherwise 1`] = `
[
  "Source code is expected to be here",
]
`;

exports[`Common tests > can trim empty lines at the start, which may change indentation otherwise 1`] = `
[
  "Source code is expected to be here",
]
`;

exports[`Common tests > handles an empty string 1`] = `{}`;
